packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -196,13 +263,55 @@ const BoldChunk = (chunks) => <Typography fontWeight="bold">{chunks}</Typography
 
 const SelectedEntriesModalContent = ({ onToggle, onConfirm, onRefresh }) => {
   const { formatMessage } = useIntl();
-  const { selectedEntries, rows, isLoading, isFetching } = useTableContext();
+  const { selectedEntries, rows, onSelectRow, isLoading, isFetching } = useTableContext();
+  const [isDialogOpen, setIsDialogOpen] = React.useState(false);
+  const [isConfirmButtonLoading, setIsConfirmButtonLoading] = React.useState(false);
+  const [rowsToDisplay, setRowsToDisplay] = React.useState([]);
 
-  const selectedEntriesWithErrorsCount = rows.filter(
+  const entriesToPublish = rows
+    .filter(({ entity, errors }) => selectedEntries.includes(entity.id) && !errors)
+    .map(({ entity }) => entity.id);
+
+  const selectedEntriesWithErrorsCount = rowsToDisplay.filter(
     ({ entity, errors }) => selectedEntries.includes(entity.id) && errors
   ).length;
   const selectedEntriesWithNoErrorsCount = selectedEntries.length - selectedEntriesWithErrorsCount;
 
+  const toggleDialog = () => setIsDialogOpen((prev) => !prev);
+
+  const handleConfirmBulkPublish = async () => {
+    try {
+      toggleDialog();
+      setIsConfirmButtonLoading(true);
+      await onConfirm(entriesToPublish);
+      setIsConfirmButtonLoading(false);
+
+      const rowsToDisplayCopy = [...rowsToDisplay];
+      const update = rowsToDisplayCopy.filter((row) => {
+        if (entriesToPublish.includes(row.entity.id)) {
+          // Deselect the entries that have been published
+          onSelectRow({ name: row.entity.id, value: false });
+        }
+
+        // Remove the entries that have been published from the table
+        return !entriesToPublish.includes(row.entity.id);
+      });
+
+      setRowsToDisplay(update);
```

Comment:
> Could we also call setRowsToDisplay in the event handler fn after awaiting onConfirm?

I believe that's what I'm doing here, right? @remidej

Reply:
Indeed, but so what use case is the effect for?

Reply:
Initalizing the state since it rows in the are context are set after fetching data from the server